docs/modules/ROOT/pages/execution-options.adoc

```diff
@@ -0,0 +1,454 @@
+= Execution Options
+:idprefix:
+:idseparator: -
+ifndef::env-github[]
+:icons: font
+endif::[]
+ifdef::env-github[]
+:caution-caption: :fire:
+:important-caption: :exclamation:
+:note-caption: :paperclip:
+:tip-caption: :bulb:
+:warning-caption: :warning:
+endif::[]
+
+JBang provides various execution options to customize how your scripts run, from JVM options to debugging and profiling features.
+
+== JVM Options and Flags
+
+=== Runtime Options
+
+Control the Java runtime with `//RUNTIME_OPTIONS` or `--runtime-option`:
+
+[source,java]
+----
+///usr/bin/env jbang "$0" "$@" ; exit $?
+//RUNTIME_OPTIONS --enable-preview -Xmx2g -Dfile.encoding=UTF-8
+
+class MyApp {
+    public static void main(String[] args) {
+        System.out.println("Max memory: " + Runtime.getRuntime().maxMemory());
+        System.out.println("File encoding: " + System.getProperty("file.encoding"));
+    }
+}
+----
+
+Or from command line:
+[source,bash]
+----
+jbang --runtime-option="-Xmx2g" --runtime-option="--enable-preview" myapp.java
+----
+
+=== Compile Options
+
+Control the Java compiler with `//COMPILE_OPTIONS` or `--compile-option`:
+
+[source,java]
+----
+///usr/bin/env jbang "$0" "$@" ; exit $?
+//COMPILE_OPTIONS --enable-preview -source 17 -Xlint:unchecked
+
+// Your code using preview features
+----
+
+Or from command line:
+[source,bash]
+----
+jbang --compile-option="--enable-preview" --compile-option="-source 17" myapp.java
+----
+
+=== Preview Features
+
+Enable Java preview features easily:
+
+[source,java]
+----
+///usr/bin/env jbang "$0" "$@" ; exit $?
+//JAVA 17+
+//PREVIEW
+
+// Use preview features here
+public class RecordExample {
+    record Point(int x, int y) {}
+    
+    public static void main(String[] args) {
+        var p = new Point(2, 4);
+        System.out.println(p);
+    }
+}
+----
+
+The `//PREVIEW` directive automatically adds the necessary compile and runtime options.
+
+== Environment Variables
+
+=== Java Options
+
+JBang respects standard Java environment variables:
+
+[source,bash]
+----
+# Java 9+ standard variables
+export JDK_JAVA_OPTIONS="-Xmx1g -Dfile.encoding=UTF-8"
+export JDK_JAVAC_OPTIONS="-Xlint:unchecked"
+
+# JBang-specific variables
+export JBANG_JAVA_OPTIONS="-Xmx2g"
+export JBANG_JAVAC_OPTIONS="-source 17"
+----
+
+=== Java Version Control
+
+[source,bash]
+----
+# Set default Java version for JBang
+export JBANG_DEFAULT_JAVA_VERSION=17
+
+# Set JDK vendor
+export JBANG_JDK_VENDOR=temurin
+
+# Run with specific version
+JBANG_DEFAULT_JAVA_VERSION=21 jbang myapp.java
```

Comment:
I'd add a `JBANG_JDK_VENDOR=...` here as well, to show you can set both in one go